buildenv/jenkins/jobs/infrastructure/artifactory_cleanup.groovy

```diff
@@ -0,0 +1,147 @@
+**/
+
+timestamps {
+    LABEL = params.LABEL
+    if (!LABEL) {
+        LABEL = 'worker'
+    }
+    node(LABEL) {
+        checkout scm
+        def variableFile = load 'buildenv/jenkins/common/variables-functions'
+        variableFile.parse_variables_file()
+        variableFile.set_artifactory_config()
+
+        def artifactory_server = params.ARTIFACTORY_SERVER ? params.ARTIFACTORY_SERVER : env.ARTIFACTORY_SERVER
+        def server = Artifactory.server artifactory_server
+        def ARTIFACTORY_SERVER_URL = server.getUrl()
+        def ARTIFACTORY_REPO = params.ARTIFACTORY_REPO ? params.ARTIFACTORY_REPO : env.ARTIFACTORY_REPO
+        def artifactoryCreds = env.ARTIFACTORY_CREDS
+        def ARTIFACTORY_NUM_ARTIFACTS = params.ARTIFACTORY_NUM_ARTIFACTS ? params.ARTIFACTORY_NUM_ARTIFACTS : env.ARTIFACTORY_NUM_ARTIFACTS
+        def ARTIFACTORY_DAYS_TO_KEEP_ARTIFACTS = params.ARTIFACTORY_DAYS_TO_KEEP_ARTIFACTS ? params.ARTIFACTORY_DAYS_TO_KEEP_ARTIFACTS : env.ARTIFACTORY_DAYS_TO_KEEP_ARTIFACTS
+
+        switch(params.JOB_TYPE) {
+            case 'TIME':
+                cleanupTime(ARTIFACTORY_SERVER_URL, ARTIFACTORY_REPO, ARTIFACTORY_DAYS_TO_KEEP_ARTIFACTS, artifactoryCreds)
+            break
+            case 'COUNT':
+                if (params.JOB_TO_CHECK){
+                    cleanupBuilds(ARTIFACTORY_SERVER_URL, ARTIFACTORY_REPO, params.JOB_TO_CHECK, ARTIFACTORY_NUM_ARTIFACTS, artifactoryCreds)
+                } else {
+                    error 'Please input a job to cleanup'
+                }
+            break
+            default:
+                error 'the JOB_TO_CHECK parameter is not properly defined. Please Enter TIME or COUNT'
+            break
+        }
+    }
+}
+
+def cleanupBuilds(artifactory_server, artifactory_repo, jobToCheck, artifactory_num_artifacts, artifactoryCreds){
+    // This parameter is originally a string and needs to be casted as an Integer
+    def artifactory_max_num_artifacts = artifactory_num_artifacts as Integer 
+
+    stage('Discover Stored Artifacts'){
+        echo "Cleaning up ${jobToCheck}"
+        echo "Keeping the latest ${artifactory_max_num_artifacts} builds"
+        currentBuild.description = "Keeping ${artifactory_max_num_artifacts} builds of ${jobToCheck}"
+
+        def request = httpRequest authentication: artifactoryCreds, consoleLogResponseBody: true, url: "${artifactory_server}/api/storage/${artifactory_repo}/${jobToCheck}"
+
+        data = readJSON text: request.getContent()
+        numberOfArtifacts = data.children.size()
+        echo "There are ${numberOfArtifacts} builds"
+    } 
+    stage('Delete Old Artifacts'){
+        def count = 0
+        if (numberOfArtifacts > artifactory_max_num_artifacts){
+            count = numberOfArtifacts - artifactory_days_to_keep_artifacts
+            def folderNames = getFolderNumbers(data.children.uri)
+            for(i=0; i  < (numberOfArtifacts - artifactory_max_num_artifacts); i++){
```

Comment:
count was created above, might as well use it or remove it

Reply:
I'm using it now and I renamed it to `﻿﻿﻿﻿﻿amount_deleted`